### PR TITLE
Feature/281

### PR DESCRIPTION
**Pull request by invpuppet**

### CODE REVIEW

**Comment by invpuppet** [comment]
[Screen_recording_20240124_065729.webm](https://github.com/H-chauvet/RISU/assets/62613130/d33528d6-5a9c-4960-b1a8-b22bbed31981)

**Comment by invpuppet** [comment]
[Screen_recording_20240124_065729.webm](https://github.com/H-chauvet/RISU/assets/62613130/e962aa0d-02d8-48e8-8b73-cdb51f1ebaf6)

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        },
+        body: jsonEncode(<String, String>{
+          'rentId': rent['id'],
+        }),
+      );
+      if (response.statusCode == 201) {
+        setState(() {
+          rent['finished'] = true;
+        });
+        print("success");
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:83]: useless debug printing

**Comment by tbellicha** [comment]
> Screen_recording_20240124_065729.webm

Wait what, there is no message telling the user that the article have been successfully given back, and the user stays on the page ?

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+      } else {
+        print('Error returnArticle(): ${response.statusCode}');
+        if (context.mounted) {
+          await MyAlertDialog.showInfoAlertDialog(
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:87]: "showInfo" while the title is "Error" ?

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                          borderRadius: BorderRadius.circular(10.0),
+                        ),
+                        padding: const EdgeInsets.all(8.0),
+                        alignment: Alignment.center,
+                        child: Text(
+                          '${rent['item']['name']} | ${rent['item']['container']['localization']}',
+                          style: TextStyle(
+                            fontSize: 18,
+                            fontWeight: FontWeight.bold,
+                            color: themeProvider.currentTheme.brightness ==
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:162]: Use the theme colors

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+                                    child: Container(
+                                      padding: const EdgeInsets.all(8.0),
+                                      color: themeProvider
+                                          .currentTheme.primaryColor,
+                                      child: Text(
+                                        'Prix',
+                                        style: TextStyle(
+                                          fontSize: 18,
+                                          fontWeight: FontWeight.bold,
+                                          color: themeProvider.currentTheme
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:192]: Use the theme colors

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+                                      padding: const EdgeInsets.all(8.0),
+                                      color: themeProvider
+                                          .currentTheme.primaryColor,
+                                      child: Text(
+                                        'Durée',
+                                        style: TextStyle(
+                                          fontSize: 18,
+                                          fontWeight: FontWeight.bold,
+                                          color: themeProvider.currentTheme
+                                                      .brightness ==
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:212]: Use the theme colors

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+                                      padding: const EdgeInsets.all(8.0),
+                                      color: themeProvider
+                                          .currentTheme.primaryColor
+                                          .withOpacity(0.6),
+                                      child: Text(
+                                        '${rent['price']} €',
+                                        style: TextStyle(
+                                          fontSize: 18,
+                                          fontWeight: FontWeight.bold,
+                                          color: themeProvider.currentTheme
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:235]: Use theme colors

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+                                      padding: const EdgeInsets.all(8.0),
+                                      color: themeProvider
+                                          .currentTheme.primaryColor
+                                          .withOpacity(0.6),
+                                      child: Text(
+                                        '${rent['duration']} heures',
+                                        style: TextStyle(
+                                          fontSize: 18,
+                                          fontWeight: FontWeight.bold,
+                                          color: themeProvider.currentTheme
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:255]: Use theme colors

File: Application/mobile/app_code/lib/pages/rent/return_state.dart [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+                  width: double.infinity,
+                  child: MyOutlinedButton(
+                    text: 'Rendre l\'article',
+                    key: const Key('rent_return-button-return_article'),
+                    onPressed: () async {
+                      bool returnRent =
+                          await MyAlertDialog.showChoiceAlertDialog(
+                        context: context,
+                        title: 'Rendre l\'article',
+                        message: 'Voules-vous rendre votre location ?',
```
Comment (tbellicha) [inline Application/mobile/app_code/lib/pages/rent/return_state.dart:286]: "Confirmer le rendu de l'article" might be better ?

File: Application/server/prisma/schema.prisma [hunk 9]
```diff
@@ -78,6 +78,7 @@ model Location {
 	userId				String
 	createdAt			DateTime	@default(now())
 	duration			Int
+	finished			Boolean		@default(false)
```
Comment (tbellicha) [inline Application/server/prisma/schema.prisma:81]: "ended" might be better ?

File: Application/server/server.js [hunk 10]
```diff
@@ -637,6 +659,89 @@ app.get('/api/rent',
...
+                  id: true,
+                  localization: true,
+                }
+              }
+            }
+          }
+        }
+      })
+      if (!rental) {
+        return res.status(401).send('location not found')
```
Comment (tbellicha) [inline Application/server/server.js:697]: Write your sentences with uppercase

File: Application/server/server.js [hunk 11]
```diff
@@ -637,6 +659,89 @@ app.get('/api/rent',
...
+  }
+)
+
+app.post('/api/rent/return',
+  passport.authenticate('jwt', { session: false }), async (req, res) => {
+    try {
+      if (!req.user) {
+        return res.status(401).send('Invalid token')
+      }
+      if (!req.body.rentId || req.params.rentId == '') {
```
Comment (tbellicha) [inline Application/server/server.js:716]: body and params ?

File: Application/server/server.js [hunk 12]
```diff
@@ -637,6 +659,89 @@ app.get('/api/rent',
...
+      }
+      return res.status(201).json({ rental: rental })
+    } catch (err) {
+      console.error(err.message)
+      return res.status(401).send('An error occurred')
+    }
+  }
+)
+
+app.post('/api/rent/return',
```
Comment (tbellicha) [inline Application/server/server.js:710]: "/api/rent/:rentId/return" would be better

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**